### PR TITLE
Add the capabilities vocabulary support

### PR DESCRIPTION
**Pull request by xuzhg**

### CODE REVIEW

**Comment by xuzhg** [comment]
Merged at: b4f019a

File: src/Microsoft.OData.Edm/Vocabularies/CapabilitiesVocabularies.xml [hunk 1]
```diff
@@ -1,18 +1,260 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Schema xmlns="http://docs.oasis-open.org/odata/ns/edm" Namespace="Org.OData.Capabilities.V1" Alias="Capabilities">
+
+  <!-- Conformance Level -->
+
+  <Term Name="ConformanceLevel" Type="Capabilities.ConformanceLevelType" AppliesTo="EntityContainer" />
```
Comment (congysu) [inline src/Microsoft.OData.Edm/Vocabularies/CapabilitiesVocabularies.xml]: This file should be the same to the one in oasis.org: http://docs.oasis-open.org/odata/odata/v4.0/errata02/os/complete/vocabularies/Org.OData.Capabilities.V1.xml

File: test/FunctionalTests/Tests/DataEdmLib/FunctionalTests/VocabularyParsingTests.cs [hunk 2]
```diff
@@ -1726,16 +1726,16 @@ public void EnumAnnotationParsingTest()
   <Term Name=""Permission"" Type=""Ns.Permissions"" Nullable=""false"" />
   <EntityContainer Name=""Container"">
     <EntitySet Name=""People"" EntityType=""Ns.Person"">
-        <Annotation Term=""Org.OData.Capabilities.V1.NavigationRestrictions"">
+        <Annotation Term=""Test.NavigationRestrictions"">
```
Comment (congysu) [inline test/FunctionalTests/Tests/DataEdmLib/FunctionalTests/VocabularyParsingTests.cs]: Why changed the namespace?

File: test/FunctionalTests/Tests/DataEdmLib/TDD/CapabilitiesVocabularyTest.cs [hunk 3]
```diff
@@ -40,32 +41,189 @@ public void TestCapabilitiesVocabularyModel()
...
+  </Term>
+  <Term Name=""UpdateRestrictions"" Type=""Capabilities.UpdateRestrictionsType"" AppliesTo=""EntitySet"">
+    <Annotation Term=""Core.Description"" String=""Restrictions on update operations"" />
+  </Term>
+  <Term Name=""DeleteRestrictions"" Type=""Capabilities.DeleteRestrictionsType"" AppliesTo=""EntitySet"">
+    <Annotation Term=""Core.Description"" String=""Restrictions on delete operations"" />
+  </Term>
 </Schema>";
 
-            var s = this.capVocModel.FindDeclaredValueTerm("Org.OData.Capabilities.V1.ChangeTracking");
```
Comment (congysu) [inline test/FunctionalTests/Tests/DataEdmLib/TDD/CapabilitiesVocabularyTest.cs]: Why removed these Asserts?

File: test/FunctionalTests/Tests/DataEdmLib/FunctionalTests/VocabularyParsingTests.cs (hunk 2 above)
Comment (xuzhg) [inline test/FunctionalTests/Tests/DataEdmLib/FunctionalTests/VocabularyParsingTests.cs]: This case is intended to test the undefined vocabulary. Before this feature, `Org.OData.Capabilities.V1.NavigationRestrictions` is an undefined vocabulary. After this feature, it now becomes a defined vocabulary in the Capabilities vocabulary model. Therefore, we should change the input for this test case. As suggestion, changed to "Test" is acceptable.

File: src/Microsoft.OData.Edm/Vocabularies/CapabilitiesVocabularies.xml (hunk 1 above)
Comment (xuzhg) [inline src/Microsoft.OData.Edm/Vocabularies/CapabilitiesVocabularies.xml]: It follows up the Core vocabulary:
https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Edm/Vocabularies/CoreVocabularies.xml
Besides, such xml is just used to build the Csdl model.

**Pull request closed without merging**